File: game/Assets/Scripts/Network/Messages/GameState.cs
using System;


namespace Network
{
    [Serializable]
    public class GameState
    {
        public Role Turn;  // Who's turn it is.
        public int TurnsRemaining;
        public int MovesRemaining;  // How many moves the player has left.
        public string GameEndDate;  // When the game ends, as a DateTime string in ISO 8601 format.
        public int Score;
    }
}

File: server/messages/game_state.py
from dataclasses import dataclass, field
from dataclasses_json import dataclass_json, config, LetterCase
from messages.rooms import Role
from datetime import datetime
from marshmallow import fields

import dateutil.parser


@dataclass_json(letter_case=LetterCase.PASCAL)
@dataclass(frozen=True)
class GameState:
    turn: Role
    turns_remaining: int
    moves_remaining: int  # Number of moves remaining.
    game_end_date: datetime = field(
        metadata=config(
            encoder=datetime.isoformat,
            decoder=dateutil.parser.isoparse,
            mm_field=fields.DateTime(format='iso')
        ))
    score: int
